js/index.js

```diff
@@ -457,7 +477,7 @@ programButton.onclick = async () => {
        
         fileArr.push({data:fileObj.data, address:offset});
     }
-    esploader.write_flash({fileArray: fileArr, flash_size: 'keep'});
+    await esploader.write_flash({fileArray: fileArr, flash_size: 'keep'});
```

Comment:
Could you try using `await esploader.write_flash(fileArr, "keep");` instead ?

The method signature is now :

```ts
fileArray: {
      data: string;
      address: number;
    }[],
    flash_size = "keep",
    flash_mode = "keep",
    flash_freq = "keep",
    erase_all = false,
    compress = true,
    reportProgress?: (fileIndex: number, written: number, total: number) => void,
    calculateMD5Hash?: (image: string) => string,
```